raiden/tests/integration/fixtures/transport.py

```diff
@@ -1,86 +1,65 @@
-MatrixTransportConfig = namedtuple('MatrixTransportConfig', 'command server')
-
-
-class TransportProtocol(Enum):
-    UDP = 'udp'
-    MATRIX = 'matrix'
+from raiden.network.transport import MatrixTransport
+from raiden.tests.fixtures.variables import TransportProtocol
+from raiden.tests.utils.transport import generate_synapse_config, matrix_server_starter
 
 
 @pytest.fixture
-def transport(request):
-    """ 'all' replaced by parametrize in conftest.pytest_generate_tests """
-    return request.config.getoption('transport')
+def public_and_private_rooms():
+    """If present in a test, conftest.pytest_generate_tests will parametrize private_rooms fixture
+    """
+    return True
 
 
-@pytest.fixture
-def transport_config(request, transport):
-    if transport == 'udp':
-        return TransportConfig(protocol=TransportProtocol.UDP, parameters=None)
-    elif transport == 'matrix':
-        command = request.config.getoption('local_matrix')
-        return TransportConfig(
-            protocol=TransportProtocol.MATRIX,
-            parameters=MatrixTransportConfig(
-                command=command,
-                server=request.config.getoption('matrix_server'),
-            ),
-        )
-    else:
-        return None
-    # can be changed with command line options, see tests/conftest.py
+@pytest.fixture(scope='session')
+def synapse_config_generator():
+    with generate_synapse_config() as generator:
+        yield generator
 
 
 @pytest.fixture
-def skip_if_not_udp(request):
-    """Skip the test if not run with UDP transport"""
-    if request.config.option.transport in ('udp', 'all'):
-        return
-    pytest.skip('This test works only with UDP transport')
+def matrix_server_count():
+    return 1
 
 
 @pytest.fixture
-def skip_if_not_matrix(request):
-    """Skip the test if not run with Matrix transport"""
-    if request.config.option.transport in ('matrix', 'all'):
+def local_matrix_servers(
+    request,
+    transport_protocol,
+    matrix_server_count,
+    synapse_config_generator,
+):
+    if transport_protocol is not TransportProtocol.MATRIX:
+        yield [None]
         return
-    pytest.skip('This test works only with Matrix transport')
 
-
-@pytest.fixture
-def public_and_private_rooms():
-    """If present in a test, conftest.pytest_generate_tests will parametrize private_rooms fixture
-    """
-    return True
+    starter = matrix_server_starter(
+        count=matrix_server_count,
+        config_generator=synapse_config_generator,
+        log_context=request.node.name,
+    )
+    with starter as server_urls:
+        yield server_urls
 
 
 @pytest.fixture
-def local_matrix_server(transport_config):
-    if not transport_config.protocol == TransportProtocol.MATRIX:
-        yield None
-        return
-
-    server = transport_config.parameters.server
+def matrix_transports(local_matrix_servers, retries_before_backoff, retry_interval, private_rooms):
```

Comment:
Very nice approach to having `matrix_transports` as a fixtures receiving `local_matrix_servers`, but may I suggest also its count to be decoupled from count of matrix_servers, being passed as a fixture as well (possibly defaulting to number of local matrix servers), and doing a roundrobin of transports around the servers, so we get more control over number of servers even if different than number if transports/clients we're setting up? e.g. 8 transports/clients spread accross 3 servers

Reply:
Good idea.
But lets improve this in a separate PR so this doesn't get any bigger than it already is.

Reply:
Made an issue so we don't forget: #3290